tests/testthat/_snaps/match.md

```diff
@@ -153,33 +181,245 @@
       Each element of `foo` can match at most 1 observation from `bar`.
       x The element at location 1 has multiple matches.
 
-# `multiple` is validated
+# `relationship` handles one-to-one case
 
     Code
-      (expect_error(vec_locate_matches(1, 2, multiple = 1.5)))
+      (expect_error(vec_locate_matches(c(2, 1), c(1, 1), relationship = "one_to_one"))
+      )
+    Output
+      <error/vctrs_error_matches_relationship_one_to_one>
+      Error in `vec_locate_matches()`:
+      ! Each element can match at most 1 observation.
+      x The element at location 2 has multiple matches.
+    Code
+      (expect_error(vec_locate_matches(c(1, 1), c(1, 2), relationship = "one_to_one"))
+      )
+    Output
+      <error/vctrs_error_matches_relationship_one_to_one>
+      Error in `vec_locate_matches()`:
+      ! Each element can match at most 1 observation.
+      x The element at location 1 has multiple matches.
+
+# `relationship` handles one-to-many case
+
+    Code
+      (expect_error(vec_locate_matches(c(1, 2, 2), c(2, 1), relationship = "one_to_many"))
+      )
+    Output
+      <error/vctrs_error_matches_relationship_one_to_many>
+      Error in `vec_locate_matches()`:
+      ! Each element can match at most 1 observation.
+      x The element at location 1 has multiple matches.
+
+# `relationship` handles many-to-one case
+
+    Code
+      (expect_error(vec_locate_matches(c(1, 2), c(1, 2, 2), relationship = "many_to_one"))
+      )
+    Output
+      <error/vctrs_error_matches_relationship_many_to_one>
+      Error in `vec_locate_matches()`:
+      ! Each element can match at most 1 observation.
+      x The element at location 2 has multiple matches.
+
+# `relationship` handles warn-many-to-many case
+
+    Code
+      (expect_warning(vec_locate_matches(c(1, 2, 1), c(1, 2, 2), relationship = "warn_many_to_many"))
+      )
+    Output
+      <warning/vctrs_warning_matches_relationship_many_to_many>
+      Warning in `vec_locate_matches()`:
+      Detected an unexpected many-to-many relationship.
+      x The element at location 2 has multiple matches.
+      x The element at location 1 has multiple matches.
```

Comment:
The many-to-many warning is another place the `needles_arg` / `haystack_arg` defaults are particularly confusing. The first line relates to `needles` and the second is for `haystack`